kong/utils.go

```diff
@@ -141,3 +143,91 @@ func VersionFromInfo(info map[string]interface{}) string {
 	}
 	return version.(string)
 }
+
+func getDefaultProtocols(schema gjson.Result) []*string {
+	var res []*string
+	fields := schema.Get("fields")
+
+	for _, field := range fields.Array() {
+		for key, value := range field.Map() {
+			if key != "protocols" {
+				continue
+			}
```

Comment:
Map() returns a map[string]Result:
https://pkg.go.dev/github.com/tidwall/gjson?utm_source=godoc#Result.Map

Please remove this loop and instead use `field.Map()["protocols"]` to do the same thing.
Linear searches are best avoided, especially when an index is already available.